src/pextlib1.0/tracelib.c

```diff
@@ -1004,28 +1016,59 @@ static int TracelibCloseSocketCmd(Tcl_Interp *interp UNUSED) {
     return TCL_OK;
 }
 
+static int pointer_strcmp(const char** a, const char** b) {
+    return strcmp(*a, *b);
+}
+
+
 static int TracelibSetDeps(Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[]) {
-    char *t, * d;
-    size_t l;
+    Tcl_Obj **objects;
+    int length;
     if (objc != 3) {
         Tcl_WrongNumArgs(interp, 2, objv, "number of arguments should be exactly 3");
         return TCL_ERROR;
     }
 
-    d = Tcl_GetString(objv[2]);
-    l = strlen(d);
-    depends = malloc(l + 2);
-    if (!depends) {
+    if (TCL_OK != Tcl_ListObjGetElements(interp, objv[2], &length, &objects)) {
+        return TCL_ERROR;
+    }
+
+    /* When called twice, do not leak memory */
+    if (depends) {
+        for (size_t i = 0; i < dependsLength; ++i) {
+            free(depends[i]);
```

Comment:
Could also set `depends[i] = NULL;` here and below, but probably not really important since the chunk of memory is unreachable afterwards.

Reply:
Thought about that but didn't precisely for that reason.